query-node/mappings/content/utils.ts

```diff
@@ -518,10 +521,54 @@ export async function unsetAssetRelations(store: DatabaseManager, dataObject: St
     })
     await store.save<Video>(video)
 
+    // check if video is fully active
+    const isFullyActive = isVideoFullyActive(video)
+
+    // update video counters for channel, channel category, and video category if needed
+    if (wasFullyActive !== isFullyActive) {
+      await updateVideoActiveCounters(store, video, isFullyActive)
+    }
+
     // emit log event
     logger.info('Content has been disconnected from Video', {
       videoId: video.id.toString(),
       dataObjectId: dataObject.id,
     })
   }
+
+  // remove data object
+  await store.remove<StorageDataObject>(dataObject)
+}
+
+export function isVideoFullyActive(video: Video): boolean {
+  return !!video.isPublic && !video.isCensored && !!video.thumbnailPhoto && !!video.media
```

Comment:
To make it consistent with https://github.com/Joystream/joystream/pull/2954, this should be:
```suggestion
  return !!video.isPublic && !video.isCensored && !!(video.thumbnailPhoto?.isAccepted) && !!(video.media?.isAccepted)
```
And there should be additional logic added in `storage_PendingDataObjectsAccepted`

Reply:
Fixed.